### DISCUSSION TITLE
Fallback via if: failure() without failing the job

### DISCUSSION THREAD

**Discussion by bfloch** [community/community#25818] [Actions]
<p>Given pseudo steps like this:</p>
<pre><code>- name: Pull docker image if available
        run: |
          docker pull A:TAG

      - name: ... else build and push Docker image
        if: failure()
        run: |
          docker build --tag A:B .
          docker push A:B
</code></pre>
<p>How do I succeed the overall job?</p>
<p>If I set continue-on-error for the first step the if: failure() will not trigger.</p>
<p>Without succeding this job I can not run other jobs with needs:</p>
<p>Thanks.</p>

**Comment by ethomson** [comment, marked as answer]
<p>A failing step will cause the workflow to fail.  Instead of allowing the step to fail and then testing for it, you can set a variable when the docker pull fails, allow that step to <em>succeed</em> and then test for that variable.</p>
<p>For example:</p>
<pre><code>- name: Pull docker image if available
        run: |
          docker pull A:TAG || echo "::set-env name=NEEDS_CONTAINER::true"
      - name: ... else build and push Docker image
        if: env.NEEDS_CONTAINER == 'true'
        run: |
          docker build --tag A:B .
          docker push A:B
</code></pre>
